client/comms/wsconn.go

```diff
@@ -236,7 +237,8 @@ func (conn *WsConn) read() {
 
 		err := ws.ReadJSON(msg)
 		if err != nil {
-			if _, ok := err.(*json.UnmarshalTypeError); ok {
+			var mErr *json.UnmarshalTypeError
+			if errors.As(err, &mErr) {
 				// JSON decode errors are not fatal, log and proceed.
 				log.Errorf("json decode error: %v", err)
```

Comment:
So should `mErr` be used here since that is the error found by errors.As in the chain of err?  We didn't use the cast error type before, but it seems appropriate.  I supposed the same Error method gets called regardless.